model/src/main/kotlin/licenses/ResolvedPackageLicenseInfo.kt

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (C) 2017-2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.model.licenses
+
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.LicenseSource
+import org.ossreviewtoolkit.model.Provenance
+import org.ossreviewtoolkit.model.config.LicenseFindingCuration
+import org.ossreviewtoolkit.model.config.PathExclude
+import org.ossreviewtoolkit.spdx.SpdxSingleLicenseExpression
+
+/**
+ * Resolved license information about a package (or project).
+ */
+data class ResolvedPackageLicenseInfo(
+    /**
+     * The identifier of the package (or project).
+     */
+    val id: Identifier,
+
+    /**
+     * The lost of [ResolvedLicenseInfo]s for this package (or project).
+     */
+    val licenses: List<ResolvedLicenseInfo>
+)
+
+/**
+ * Resolved information for a single license.
+ */
+data class ResolvedLicenseInfo(
+    /**
+     * The license.
+     */
+    val license: SpdxSingleLicenseExpression,
+
+    /**
+     * The sources where this license was found.
+     */
+    val sources: Set<LicenseSource>,
+
+    /**
+     * The original declared license, if this license is a processed declared license, or null.
+     */
+    val originalDeclaredLicense: String?,
+
+    /**
+     * All text locations where this license was found.
+     */
+    val locations: Set<ResolvedTextLocation>,
+
+    /**
+     * All copyrights there were associated to this license.
+     */
+    val copyrights: Set<ResolvedCopyrightFinding>
+)
+
+/**
+ * A resolved text location.
+ */
+data class ResolvedTextLocation(
+    /**
+     * The provenance of the file.
+     */
+    val provenance: Provenance,
```

Comment:
It's important to associate the provenance with the findings, because they could potentially come from different sources.

Reply:
I think I was more after (1) whether its associated with the small findings object or whether there is a larger object (like Findings) holding the provenance and (2) that provenance actually contains unneeded information like the `downloadTime`.

Anyhow, I think it can be easily changed later (if it makes sense) so let's keep it as is to not loose focus.